src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/DoNotInitializeUnnecessarily.cs

```diff
@@ -37,7 +37,11 @@ public override void Initialize(AnalysisContext analysisContext)
             {
                 var init = (IFieldInitializerOperation)context.Operation;
                 IFieldSymbol? field = init.InitializedFields.FirstOrDefault();
-                if (field != null && !field.IsConst && init.Value != null && UsesKnownDefaultValue(init.Value, field.Type))
+                if (field != null &&
+                    !field.IsConst &&
+                    init.Value != null &&
+                    field.GetAttributes().Length == 0 && // in case of attributes that impact nullability analysis
```

Comment:
We can consider to bail out only if attribute has "Null" within its name? I am fine with current conservative approach as well.

Reply:
Hadn't concerned the naming aspect.  It's a good suggestion.  At the same time, I think the conservative place is a good place to start here: worst case, we miss a few.  And the rule is only "info" by default, which I think (correct me if I'm wrong) gives more flexibility in the future should we find more opportunities to warn.

Reply:
Yep, starting conservative is fine.